src/api/File.js

```diff
@@ -78,29 +80,81 @@ class File extends Item {
         this.successHandler(`${downloadBaseUrl}?${downloadUrlQuery}`);
     }
 
+    /**
+     * Polls an item's infoUrl until it's representation is generated or an error occurrs.
+     * Calls callback if representation was successfully generated.
+     *
+     * @param {BoxItem} item - BoxItem that needs infoUrl polled
+     * @param {string} thumbanilUrl - thumbanil url of item
+     * @param {string, string) => void} callback - function to call if polling results in success
+     * @return {Promise<void>}
+     */
+    async pollInfoUrl(
+        item: BoxItem,
+        thumbnailUrl: string,
+        callback: (id: string, thumbnailUrl: string) => void = noop,
+    ): Promise<void> {
+        const { id } = item;
+        const infoUrl = getProp(item, 'representations.entries[0].info.url');
+
+        const numOfTries = 8;
+        const initialTimeout = 1000;
+        const backoffFactor = 2;
```

Comment:
I think this should equate to waiting a maximum of approximately `2^8 = 256` seconds for a representation to generate.